pkg/pluginmanager/manager.go

```diff
@@ -750,42 +748,22 @@ func installOrUpgradePlugin(p *discovery.Discovered, version string, installTest
 
 	// Initialize the spinner if the spinner is allowed
 	if component.IsTTYEnabled() {
-		// Create a channel to receive OS signals
-		signalChannel := make(chan os.Signal, 1)
-		// Register the channel to receive interrupt signals (e.g., Ctrl+C)
-		signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
-		defer func() {
-			signal.Stop(signalChannel)
-			close(signalChannel)
-		}()
-
 		// Initialize the spinner
 		spinner = component.NewOutputWriterSpinner(component.WithOutputStream(os.Stderr),
 			component.WithSpinnerText(installingMsg),
-			component.WithSpinnerStarted(),
-			component.WithSpinnerFinalText(installedMsg, log.LogTypeINFO))
-
+			component.WithSpinnerStarted())
+		spinner.SetFinalText(errMsg, log.LogTypeERROR)
 		defer spinner.StopSpinner()
-
-		// Start a goroutine that listens for interrupt signals
-		go func(s component.OutputWriterSpinner) {
-			sig := <-signalChannel
-			if sig != nil {
-				if s != nil {
-					s.SetFinalText(errMsg, log.LogTypeERROR)
-					s.StopSpinner()
-				}
-				os.Exit(128 + int(sig.(syscall.Signal)))
-			}
-		}(spinner)
 	} else {
 		log.Info(installingMsg)
 	}
 
 	pluginErr := verifyInstallAndInitializePlugin(plugin, p, version, installTestPlugin)
-	if pluginErr != nil {
-		if spinner != nil {
+	if spinner != nil {
+		if pluginErr != nil {
 			spinner.SetFinalText(errMsg, log.LogTypeERROR)
```

Comment:
Considering L755 already sets the error message as the final text, do we need to set it again here on error?
Will it be possible to simplify this with just as mentioned in the previous comment?

```
        pluginErr := verifyInstallAndInitializePlugin(plugin, p, version, installTestPlugin)
	if pluginErr == nil && spinner != nil {
        	spinner.SetFinalText(installedMsg, log.LogTypeINFO)
	}
```

Reply:
good catch, thanks. Updated.